util/imageutil/config.go

```diff
@@ -213,10 +214,25 @@ func childrenConfigHandler(provider content.Provider, platform platforms.MatchCo
 			}
 
 			if platform != nil {
-				for _, d := range index.Manifests {
-					if d.Platform == nil || platform.Match(*d.Platform) {
-						descs = append(descs, d)
-					}
+				if len(index.Manifests) > 0 {
```

Comment:
Instead of this, shouldn't we do:

```
children := childrenConfigHandler(cache, platform)
children = images.LimitManifests(children, platform, 1)
```

Reply:
Fixed